projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -168,6 +191,23 @@ export class ExplorerComponent {
       this.initialState$.pipe(map(value => value.contextToggle.value.dashboardContext)),
       this.contextChangeSubject
     );
+    this.featureStateResolver.getFeatureState(ApplicationFeature.SavedQueries).subscribe(featureState => {
+      this.enableSavedQueries = featureState === FeatureState.Enabled ? true : false;
+    });
+  }
+
+  public onClickSaveQuery(): void {
+    this.preferenceService.get('savedQueries', []).subscribe(queries => {
+      this.savedQueries = queries as SavedQuery[];
+    });
+
+    const currentScope =
+      this.filters[0]?.metadata.scope === SPAN_SCOPE ? ScopeQueryParam.Spans : ScopeQueryParam.EndpointTraces;
+    const currentFilters = this.filters.map(filter => filter.urlString);
+
+    this.savedQueries.push({ scope: currentScope, filters: currentFilters });
+    this.preferenceService.set('savedQueries', this.savedQueries);
```

Comment:
If the `set` is an observable action, we should probably do the `toast` in the subscribe part? That way in case something goes wrong setting it, we wouldn't show the notification.

Reply:
There's no subscribe() available on the set() method:
<img width="895" alt="Screenshot 2022-05-13 at 2 39 58 PM" src="https://user-images.githubusercontent.com/29686866/168251423-00166e02-30a3-4dd4-b660-2aed084ae85e.png">

Should I add a try-catch here instead?

Reply:
I guess we can let it be in that case. A `try....catch` block might not be needed, at least for in browser storage.
When we revise to have storage over DB using some network call, then we might need it, though it maybe within `PreferenceService`.